src/repo.rs

```diff
@@ -59,6 +59,24 @@ impl Repository {
         }
     }
 
+    /// Find and open a repository, with additional options.
+    pub fn open_ext<P: AsRef<Path>>(path: P,
+                                    flags: RepositoryOpenFlags,
+                                    ceiling_dirs: Option<&str>)
```

Comment:
There's `iter2cstrs`, but that just constructs a `git_strarray`.  `git_repository_open_ext` takes a single string of paths separated by `GIT_PATH_LIST_SEPARATOR`.

Also, if the caller already has this parameter in the form libgit2 expects, it seems unfortunate to have to split it just to have git2-rs re-join it.

I think it makes more sense for the caller to call `std::env::join_paths` themselves, rather than having `open_ext` do it for them; that way, if you already have a pre-joined path in the form you'd get it from `$GIT_CEILING_DIRECTORIES`, you can just pass that.